packages/firebase_crashlytics/firebase_crashlytics/example/lib/main.dart

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:firebase_core/firebase_core.dart';
+import 'package:firebase_crashlytics/firebase_crashlytics.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+// Toggle this to cause an async error to be thrown during initialization
+// and to test that runZonedGuarded() catches the error
+final _kShouldTestAsyncErrorOnInit = false;
+
+// Toggle this for testing Crashlytics in your app locally.
+final _kTestingCrashlytics = true;
+
+main() {
+  WidgetsFlutterBinding.ensureInitialized();
+  runZonedGuarded(() {
+    runApp(MyApp());
+  }, (error, stackTrace) {
+    print('runZonedGuarded: Caught error in my root zone.');
+    FirebaseCrashlytics.instance.recordError(error, stackTrace);
+  });
+}
+
+// ignore: public_member_api_docs
+class MyApp extends StatefulWidget {
+  @override
+  _MyAppState createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  final GlobalKey<ScaffoldState> _scaffoldKey = GlobalKey<ScaffoldState>();
+
+  // Set default `_initialized` and `_error` state to false
+  bool _initialized = false;
+  bool _error = false;
+
+  void testAsyncErrorOnInit() async {
+    Future<void>.delayed(const Duration(seconds: 2), () {
+      final List<int> list = <int>[];
+      print(list[100]);
+    });
+  }
+
+  // Define an async function to initialize FlutterFire
+  void initializeFlutterFire() async {
+    try {
+      // Wait for Firebase to initialize and set `_initialized` state to true
+      await Firebase.initializeApp();
+
+      if (_kTestingCrashlytics) {
+        // Force enable crashlytics collection enabled if we're testing it.
+        await FirebaseCrashlytics.instance
+            .setCrashlyticsCollectionEnabled(true);
+      } else {
+        // Else only enable it in non-debug builds.
+        // You could additionally extend this to allow users to opt-in.
+        await FirebaseCrashlytics.instance
+            .setCrashlyticsCollectionEnabled(!kDebugMode);
+      }
+
+      // Pass all uncaught errors to Crashlytics.
+      Function originalOnError = FlutterError.onError;
+      FlutterError.onError = (FlutterErrorDetails errorDetails) async {
+        await FirebaseCrashlytics.instance.recordFlutterError(errorDetails);
+        // Forward to original handler.
+        originalOnError(errorDetails);
+      };
+
+      setState(() {
+        _initialized = true;
+      });
+      if (_kShouldTestAsyncErrorOnInit) testAsyncErrorOnInit();
+    } catch (e) {
+      // Set `_error` state to true if Firebase initialization fails
+      print(e);
+      setState(() {
+        _error = true;
+      });
+    }
+  }
+
+  @override
+  void initState() {
+    initializeFlutterFire();
```

Comment:
The convention afaik is to generally call the `super.initState` so wondering if there's a reason it's last in this case